node/src/components/small_network/message.rs

```diff
@@ -2,16 +2,25 @@ use std::fmt::{self, Debug, Display, Formatter};
 
 use serde::{Deserialize, Serialize};
 
+use crate::crypto::hash::Digest;
+
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub enum Message<P> {
-    Handshake { network_name: String },
+    Handshake {
+        genesis_config_hash: Digest,
+        #[serde(default)]
+        network_name: String,
```

Comment:
In the description of this PR you say that the old delta node (running version `0.7.6`) will ignore the new field name `genesis_config_hash`. How does this happen? Programmatically I mean. What type of handshake would a `0.8.0` node send to `0.7.6`? It will include both _a_ config hash and a network name. How do `0.7.6` nodes know to ignore the network name?

Reply:
Bincode's `deserialize()` function [sets the config option to allow trailing bytes](https://github.com/bincode-org/bincode/blob/trunk/src/lib.rs#L183) which means the 0.7.6 nodes just ignore the unused bytes comprising the network name.

It's a little fragile in that we need to ensure the fields we want ignored are at the end of the struct.  But this is temporary code for delta only I believe.

Reply:
Ha, nice hack! and it's not even in _our code_!